Adding performance message for Table.read with `memmap=True` (#9404)
astropy/astropy: docs/table/performance.inc.rst

It uses the non-masked version of data and it is faster::

    >>> tm.write('tm.ecsv', overwrite=True, serialize_method='data_mask') # doctest: +SKIP

> Reading a ``.fits`` in |Table| using

Reviewer: Should be `Read a large FITS file in ...`, and same further down `read a large FITS file`.  For small files using `memmap=True` will likely degrade performance.

Reviewer: The text still refers to reading a `.fits`, that needs to be `FITS file`.

:meth:`~astropy.table.Table.read` when ``memmap=False`` can be very slow
because it reads full |Table| in the memory. When you want to 
read ``.fits`` using :meth:`~astropy.table.Table.read`,then use ``memmap=True``.
It does not load the data, but as soon as a column data is used the 
whole data is loaded :

.. doctest-skip::

    >>> import numpy as np
    >>> from astropy.table import Table

>     >>> tbl = Table({ 'a': np.arange(1e7), 'b': np.arange(1e7, dtype=float), 'c': np.arange(1e7, dtype=float)})

Reviewer: This line is too long, best to break into 3 lines for visual clarity.

>     >>> tbl.write(tmpfile, overwrite=True)

Reviewer: Use `'test.fits'` not `tmpfile` here to match the next line.

It uses the non-masked version of data and it is faster::

    >>> tm.write('tm.ecsv', overwrite=True, serialize_method='data_mask') # doctest: +SKIP

Reading a large ``.fits`` in |Table| using
:meth:`~astropy.table.Table.read` when ``memmap=False`` can be very slow
because it reads full |Table| in the memory. When you want to read a 

> large ``.fits`` using :meth:`~astropy.table.Table.read`,then use ``memmap=True``.

Reviewer 2: missing a whitespace after the `,`

Author: Done Thanks @MSeifert04

It uses the non-masked version of data and it is faster::

    >>> tm.write('tm.ecsv', overwrite=True, serialize_method='data_mask') # doctest: +SKIP

Reading a ``.fits`` in |Table| using
:meth:`~astropy.table.Table.read` when ``memmap=False`` can be very slow
because it reads full |Table| in the memory ::

    >>> import numpy as np
    >>> from astropy.table import Table

>     >>> tbl = Table({

Reviewer 3: Should we skip this too? CI has limits.

Reviewer 4: agreed with @pllim. For this example I think adding `.. doctest-skip::` directive in front of the example would work well. See example in e.g. docs/nddata/utils.rst

>     >>> tbl = Table({

Reviewer 4: Could you please reformat to an un-black way? I know it's fashionable tool right now, but I don't find having the closing parens in a new line, and overall spreading stuff over more lines etc making anything more readable.

    ...     'a': np.arange(1e7),
    ...     'b': np.arange(1e7, dtype=float),
    ...     'c': np.arange(1e7, dtype=float),
    ...  })
    >>> tbl.write(tmpfile, overwrite=True) # doctest: +SKIP

>     >>> t = Table.read('test.fits', memmap=False) # doctest: +SKIP

Reviewer 4: I don't think we need to show to slow way. So I would suggest to merge the two paragraph into one, as well as have only one example.

It uses the non-masked version of data and it is faster::

    >>> tm.write('tm.ecsv', overwrite=True, serialize_method='data_mask') # doctest: +SKIP

> Read FILE with memmap=True

Reviewer 5: FILE -> FITS

--------------------------

By default :meth:`~astropy.table.Table.read` will read the whole table into memory, which 

> will take a lot of memory and can take a lot of time, depending on the table size and 

Reviewer 5: will -> can

> file format. In some cases, it's possibly to only read a subset of the table by choosing 

Reviewer 6: ```suggestion
file format. In some cases, it's possible to only read a subset of the table by choosing 
```

Reviewer 6: It might be useful to clarify the "In some cases". I *think* that the table will always be loaded fully in memory if one column must be converted to another format, i.e. when there is a boolean column, or when some column must be scaled. 
This is if I remember correctly the discussions on @astrofrog 's PR, so maybe he remembers better than me ;).

> file format. In some cases, it's possible to only read a subset of the table by choosing 

Reviewer 7: ```suggestion
file format. In some cases, it is possible to only read a subset of the table by choosing 
```

the option ``memmap=True``.

For FITS binary tables, the data is stored row by row, and it is possible to read only a 

> subset of rows, but reading a full column load the whole table data into memory:

Reviewer 7: ```suggestion
subset of rows, but reading a full column loads the whole table data into memory:
```

.. doctest-skip::

    >>> import numpy as np
    >>> from astropy.table import Table
    >>> tbl = Table({ 'a': np.arange(1e7),
    ...               'b': np.arange(1e7, dtype=float),
    ...               'c': np.arange(1e7, dtype=float)})
    >>> tbl.write('test.fits', overwrite=True)

>     >>> table = Table.read("test.fits", memmap=True) # Very fast, doesn't actually load data

Reviewer 6: Nitpick but it's better to be consistent with the other lines:
```suggestion
    >>> table = Table.read('test.fits', memmap=True) # Very fast, doesn't actually load data
```

    >>> table2 = tbl[:100] #  Fast, will read only first 100 rows
    >>> print(table2) # accessing column data triggers the read
     a    b    c  
    ---- ---- ----
    0.0  0.0  0.0
    1.0  1.0  1.0
    2.0  2.0  2.0
    ...  ...  ...
    98.0 98.0 98.0
    99.0 99.0 99.0
    Length = 100 rows

>     >>> col = table["my_column"] # Will load all table into memory

Reviewer 6: ```suggestion
    >>> col = table['my_column'] # Will load all table into memory
```

> Right now :meth:`~astropy.table.Table.read` does not supports ``memmap=True`` for HDF5 and ASCII 

Reviewer 5: "Right now" -> "At the moment"

Reviewer 5: supports -> support

Reviewer 5: for HDF5 -> for the HDF5
